### PR TITLE
Refactor OO section part 2

### PR DESCRIPTION
**Pull request by Girgias**
Some misses from the first time and doing some drive by fixes.

### CODE REVIEW

File: language/oop5/autoload.xml [hunk 1]
```diff
@@ -14,31 +14,27 @@
...
     Autoloading is not available if using PHP in CLI
     <link linkend="features.commandline">interactive mode</link>.
    </para>
   </note>
-  <note>
+  <warning>
+   <!-- Is this still true? -->
    <para>
     If the class name is used e.g. in <function>call_user_func</function> then
     it can contain some dangerous characters such as <literal>../</literal>.
```
Comment (afilina) [inline language/oop5/autoload.xml:35]: I don't see how one can supply `../` as the class name here. I don't see a test for it and I can't reproduce the behavior. Perhaps this refers to some very old bug. Without a code snipper demonstrating it, that warning doesn't really make any sense to me.

File: language/oop5/magic.xml [hunk 2]
```diff
@@ -27,11 +27,21 @@
    with them.
   </para>
 
-  <note>
+  <warning>
    <simpara>
-    All magic methods <emphasis>MUST</emphasis> be declared as <literal>public</literal>
+    All magic methods <emphasis>must</emphasis> be declared as
+    <literal>public</literal>, otherwise an <constant>E_WARNING</constant>
+    is emitted. Prior to PHP 8.0.0, no diagnostic was emitted.
```
Comment (afilina) [inline language/oop5/magic.xml:34]: This doesn't seem to be the case even in 8.0.0: https://3v4l.org/hW7eN

**Review by afilina — CHANGES_REQUESTED** [review]
Partial review. Will check the rest later today.

File: language/oop5/magic.xml (hunk 2 above)
Comment (Girgias) [reply language/oop5/magic.xml:34]: No visibility === public visibility, however it seems a diagnostic depends on the magic methods, amazing... (see https://3v4l.org/PS0lf and https://3v4l.org/q5N43)

File: language/oop5/static.xml [hunk 3]
```diff
@@ -18,9 +18,8 @@
 
   <para>
    Declaring class properties or methods as static makes them accessible
-   without needing an instantiation of the class. A property declared as
-   static cannot be accessed with an instantiated class object (though
-   a static method can).
+   without needing an instantiation of the class.
+   These can also be accessed within an instantiated class object.
```
Comment (afilina) [inline language/oop5/static.xml:22]:
```suggestion
   These can also be accessed statically within an instantiated class object.
```

Not sure whether the distinction is needed here.

**Review by afilina — CHANGES_REQUESTED** [review]

File: language/oop5/autoload.xml (hunk 1 above)
Comment (salathe) [reply language/oop5/autoload.xml:35]: The note was added in 2007 in response to https://bugs.php.net/42744.  PHP added validation to the class name in 2014 (php/php-src@2d31eadbbf147a157cb4a0c89adaf30fee7371f0, PHP 5.4.24 and PHP 5.5.8), which rendered the note obsolete for folks using those or newer versions of PHP.

Comment (afilina) [reply language/oop5/autoload.xml:35]: Thanks for finding this out, @salathe. I was looking for the error in phpt files starting with `call_user_func*`, heh.

File: language/oop5/magic.xml [hunk 4]
```diff
@@ -27,11 +27,21 @@
...
    </simpara>
-  </note>
+  </warning>
+  <warning>
+   <para>
+    If type declarations are used in the definition of a magic method, they
+    must be identical to the signature described in this document.
+    Otherwise a fatal error is emitted.
```
[the remark this answers is not in the document]
Comment (Girgias) [reply language/oop5/magic.xml:41]: There isn't a rule, and I probably just copy pasted it from code output. But yeah not great to be inconsistent.

File: language/oop5/static.xml [hunk 5]
```diff
@@ -29,16 +28,16 @@
...
 
    <warning>
     <para>
      Calling non-static methods statically throws an <classname>Error</classname>.
     </para>
     <para>
-     Prior to PHP 8.0.0, calling non-static methods statically is deprecated, and will
-     generate an <constant>E_DEPRECATED</constant> warning.
+     Prior to PHP 8.0.0, calling non-static methods statically is deprecated, and
+     generated an <constant>E_DEPRECATED</constant> warning.
```
Comment (Crell) [inline language/oop5/static.xml:40]: The verb tenses here are hurting my brain.  It really feels like "prior to 8.. were deprecated..." is more correct.  I can see the reason for using "is", but I really think "were" is more correct.

```suggestion
     Prior to PHP 8.0.0, calling non-static methods statically were deprecated, and
     generated an <constant>E_DEPRECATED</constant> warning.
```

Comment (Girgias) [reply language/oop5/static.xml]: Fixed and I think it makes more sense indeed.

Comment (cmb69) [reply language/oop5/static.xml]: Not a native speaker, but the plural looks wrong to me (the past tense looks right, though present perfect may be even more appropriate).

**Pull request closed without merging**